docs/architecture/adr-002-ibc-relayer.md

```diff
@@ -0,0 +1,644 @@
+
+#### Data Availability
+The relayer monitors the chain state to determine when packet forwarding is required. The relayer must be able to retrieve the data within some time bound. This is referred to as **data availability**.
+
+#### Data Legibility
+IBC protocol defines the minimal data set that must be made available to relayers for correct operation of the protocol. The relayer expects the data to be legible, i.e. **data should be serialized** according to the IBC specification format; this includes consensus state, client, connection, channel, and packet information, and any auxiliary state structure necessary to construct proofs of inclusion or exclusion of particular key/value pairs in state. 
+ - [IBC Specification] some protobuf specifications can be found under individual ICS-es, for exmple [ICS-03 connection protobufs](https://github.com/cosmos/ics/blob/master/spec/ics-002-client-semantics/data-structures.proto)
+
+#### IBC Querier Module
+IBC host state machines MUST provide query functions for all data available to relayers. For Cosmos/Tendermint chains this means:
+- the IBC modules on chain correctly implement and respond to queries
+  - [IBC-Modules-Rust] an implementation for some queries currently exist in Cosmos-SDK and same and more need to be implemented in Rust. The full requirements are detailed in section Relayer Queries. 
+- the relayer needs the ability to send rpc/http ABCI queries to and receive replies from Tendermint/Cosmos-SDK
+  - [[ABCI Rust](https://github.com/tendermint/rust-abci)] - ABCI Rust implementation
+  - [IBC-Modules-Rust] identifier validation is required (ICS-024)
+  - [IBC-Modules-Rust] requires Rust types for all query responses
+  - [[Merkle-Proofs-Rust](https://github.com/confio/ics23/tree/master/rust)] (candidate implementation) - some query responses include proofs and included in IBC transactions by the relayer (some may be validated, TBD)
+
+#### IBC Messages
+The relayer creates transactions that include IBC messages to manage clients, connections and channels, and send application packets to destination chains. These messages must be defined in the IBC Rust implementation [IBC-Modules-Rust].
+
+#### IBC Logging System
+IBC packet data & timeouts are not stored directly in the chain state (as this storage is presumed to be expensive) but are instead committed to with a succinct cryptographic commitment (only the commitment is stored).
+As a consequence, IBC requires that a **host state machine MUST provide an event logging system** that logs data in the course of transaction execution. **Logs must be queryable** by relayers to read IBC packet data & timeouts. 
+
+The logging system must provide the following functions:
+ - [IBC-Modules-Go] emitLogEntry for emitting log entries called by the state machine during transaction execution:
```

Comment:
Why reference to Go?

Reply:
in the initial version we will test with Cosmos-SDK/TM in Go chains.